_posts/2022-09-08-storage-write-removal-before-conditional-termination.md

```diff
@@ -37,6 +37,8 @@ More concretely, the bug can occur if a contract contains the following pattern:
     - It overwrites the storage write in (1).
     - It reverts.
 
+If the initial storage write could possibly be read back (directly or in any internal or external call) between (1) and (3), it is, of course, never removed. (Note, however, that in presence of the optimizer it is not always easy to determine if a storage read at the Solidity level will actually translate to a load instruction in assembly. For example the [Load Resolver](https://docs.soliditylang.org/en/latest/internals/optimizer.html#load-resolver) step can use the knowledge of values that were written before by the contract to replace a `sload()` directly with the value that it would read.)
```

Comment:
Why the parenthesis though? I think it would work just fine without one and overall the text is more readable if it's less nested (just like code incidentally :P).

Reply:
I'm afraid that all those rather unrelated details become to distracting from the core issue at some point :-). I was half-serious above when I said I'd like footnotes :-D. If we had them, I'd make this one - parentheses were a half-measure towards that... but yeah, probably not really helpful and rather making things worse, I can remove them again.

Reply:
I think it's fine without parenthesis. It would only make sense if we could really have a footnote. Or a background section that would explain Load Resolver :) As is, I'd leave as a normal part of the paragraph.

Reply:
Yep, did that.